applications/txn_plant/targets/freertos.armv7m.ek-tm4c123gxl/main.cxx

```diff
@@ -0,0 +1,278 @@
+    stack.node(), kOutputGpio, ARRAYSIZE(kOutputGpio), cfg.seg().consumers());
+
+// Similar syntax for the producers.
+openlcb::ConfiguredProducer producer_sw1(
+    stack.node(), cfg.seg().producers().entry<0>(), SW1_Pin());
+openlcb::ConfiguredProducer producer_sw2(
+    stack.node(), cfg.seg().producers().entry<1>(), SW2_Pin());
+openlcb::ConfiguredProducer producer_sw3(
+    stack.node(), cfg.seg().producers().entry<2>(), (const Gpio*)&the_button);
+
+// The producers need to be polled repeatedly for changes and to execute the
+// debouncing algorithm. This class instantiates a refreshloop and adds the two
+// producers to it.
+openlcb::RefreshLoop loop(
+			  stack.node(), {producer_sw1.polling(), producer_sw2.polling(), producer_sw3.polling()});
+
+
+class Configuration : private DefaultConfigUpdateListener {
+private:
+
+  /// Notifies the component that there is new configuration available for
+  /// loading.
+  ///
+  /// The call is made on the main executor, so the call must not
+  /// block. Reading the given EEPROM device should be fine. Asynchronous
+  /// operations may be implemented by using a special return status RETRY:
+  /// the runner will call the same method on the same component once more
+  /// after the done callback is invoked. This allows implementing state
+  /// machines.
+  ///
+  /// @param fd is the file descriptor for the EEPROM file. The current
+  /// offset in this file is unspecified, callees must do lseek.
+  /// @param initial_load is true if this is the first load upon starting the
+  /// binary.
+  /// @param done must be notified when the call and its dependent actions
+  /// are complete. No other configuration component will be called until the
+  /// done callback is invoked.
+  ///
+  /// @return any necessary action. If returns UPDATED, then assumes that the
+  /// configuration change was applied. If returns RETRY, then the same call
+  /// will be made again after the notifiable is called. If return
+  /// REINIT_NEEDED or REBOOT_NEEDED then at the end of the configuration
+  /// update process the node will be reinitialized or rebooted accordingly.
+  UpdateAction apply_configuration(int fd, bool initial_load,
+				   BarrierNotifiable *done) {
+    done->notify();
+    if (initial_load)
+      return UPDATED;
+    
+    return REBOOT_NEEDED;
+  } // apply_configuration
+  
+  /// Clears configuration file and resets the configuration settings to
+  /// factory value.
+  ///
+  /// @param fd is the file descriptor for the EEPROM file. The current
+  /// offset in this file is unspecified, callees must do lseek.
+  void factory_reset(int fd)
+  {
+    for( int i=0; i<NUM_INPUTS;++i) {
+      //char tmp[10];
+      //string str;
+      //unsigned_integer_to_buffer(i, tmp);
+      //cfg.seg().producers().entry(i).description().write(fd,tmp);
+      cfg.seg().producers().entry(i).debounce().write(fd,3);
+      cfg.seg().producers().entry(i).event_on().write(fd,(NODE_ID << 16) + 8+(i*2) );
```

Comment:
You have two choices about the event IDs here. What you did here is fine if you want a fixed layout of event IDs for your node. This is not recommended for general purpose nodes; in the standard we recommend always new event IDs to be assigned when a factory reset is performed. I have code for that:
```
stack.factory_reset_all_events(cfg.seg().internal_config(), fd);
```
Note you still need the write call for the debounce parameter.